packages/manifold/contracts/gachaclaims/IGachaLazyClaim.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+/**
+ * Gacha Lazy Claim interface
+ */
+interface IGachaLazyClaim {
+  enum StorageProtocol {
+    INVALID,
+    NONE,
+    ARWEAVE,
+    IPFS
+  }
+
+  error InvalidStorageProtocol();
+  error InvalidDate();
+  error InvalidInstance();
+  error InvalidInput();
+  error InvalidPayment();
+  error InvalidSignature();
+  error InvalidMintCount();
+  error InvalidVariationIndex();
+  error InvalidStartingTokenId();
+  error ClaimAlreadyInitialized();
+  error ClaimNotInitialized();
+  error ClaimInactive();
+  error ClaimSoldOut();
+  error TokenDNE();
+  error FailedToTransfer();
+  error TooManyRequested();
+  error CannotLowerTotalMaxBeyondTotal();
+  error CannotChangeTokenVariations();
+  error CannotChangePaymentToken();
+  error CannotLowertokenVariationsBeyondVariations();
+  error CannotMintMoreThanReserved();
+  error CannotMintFromContract();
+
+  event GachaClaimInitialized(address indexed creatorContract, uint256 indexed instanceId, address initializer);
+  event GachaClaimUpdated(address indexed creatorContract, uint256 indexed instanceId);
+  event GachaClaimMintReserved(
+    address indexed creatorContract,
+    uint256 indexed instanceId,
+    address indexed collector,
+    uint32 mintCount
+  );
+
+  struct VariationMint {
+    uint8 variationIndex;
+    uint32 amount;
+    address recipient;
```

Comment:
minor: if my understanding is correct -- should we be swapping the ordering for struct packing into 2 storage slots instead of 3 here?

eg:
```suggestion
    uint8 variationIndex;
    address recipient;
    uint32 amount;
```